src/measure_theory/integral/circle_integral.lean

```diff
@@ -0,0 +1,513 @@
+  $\int_{0}^{2π}(c + Re^{θ i})' f(c+Re^{θ i})\,dθ$.
+
+## Tags
+
+integral, circle, Cauchy integral
+-/
+
+variables {E : Type*} [measurable_space E] [normed_group E]
+
+noncomputable theory
+
+open_locale real nnreal interval pointwise topological_space
+open complex measure_theory topological_space metric function set filter asymptotics
+
+/-!
+### `circle_map`, a parametrization of a circle
+-/
+
+/-- The exponential map $θ ↦ c + R e^{θi}$. The range of this map is the circle in `ℂ` with center
+`c` and radius `|R|`. -/
+def circle_map (c : ℂ) (R : ℝ) : ℝ → ℂ := λ θ, c + R * exp (θ * I)
+
+/-- `circle_map` is `2π`-periodic. -/
+lemma periodic_circle_map (c : ℂ) (R : ℝ) : periodic (circle_map c R) (2 * π) :=
+λ θ, by simp [circle_map, add_mul, exp_periodic _]
+
+@[simp] lemma circle_map_sub_center (c : ℂ) (R : ℝ) (θ : ℝ) :
+  circle_map c R θ - c = circle_map 0 R θ :=
+by simp [circle_map]
+
+@[simp] lemma abs_circle_map_zero (R : ℝ) (θ : ℝ) : abs (circle_map 0 R θ) = |R| :=
+by simp [circle_map]
+
+lemma circle_map_mem_sphere' (c : ℂ) (R : ℝ) (θ : ℝ) : circle_map c R θ ∈ sphere c (|R|) :=
+by simp
+
+lemma circle_map_mem_sphere (c : ℂ) {R : ℝ} (hR : 0 ≤ R) (θ : ℝ) : circle_map c R θ ∈ sphere c R :=
+by simpa only [_root_.abs_of_nonneg hR] using circle_map_mem_sphere' c R θ
+
+lemma circle_map_mem_closed_ball (c : ℂ) {R : ℝ} (hR : 0 ≤ R) (θ : ℝ) :
+  circle_map c R θ ∈ closed_ball c R :=
+sphere_subset_closed_ball (circle_map_mem_sphere c hR θ)
+
+/-- The range of `circle_map c R` is the circle with center `c` and radius `|R|`. -/
+@[simp] lemma range_circle_map (c : ℂ) (R : ℝ) : range (circle_map c R) = sphere c (|R|) :=
+calc range (circle_map c R) = c +ᵥ R • range (λ θ : ℝ, exp (θ * I)) :
+  by simp only [← image_vadd, ← image_smul, ← range_comp, vadd_eq_add, circle_map, (∘), real_smul]
+... = sphere c (|R|) : by simp [smul_sphere R (0 : ℂ) zero_le_one, real.norm_eq_abs]
+
+/-- The image of `(0, 2π]` under `circle_map c R` is the circle with center `c` and radius `|R|`. -/
+@[simp] lemma image_circle_map_Ioc (c : ℂ) (R : ℝ) :
+  circle_map c R '' Ioc 0 (2 * π) = sphere c (|R|) :=
+by rw [← range_circle_map, ← (periodic_circle_map c R).image_Ioc real.two_pi_pos 0, zero_add]
+
+@[simp] lemma circle_map_eq_center_iff {c : ℂ} {R : ℝ} {θ : ℝ} : circle_map c R θ = c ↔ R = 0 :=
+by simp [circle_map, exp_ne_zero]
+
+@[simp] lemma circle_map_zero_radius (c : ℂ) : circle_map c 0 = const ℝ c :=
+funext $ λ θ, circle_map_eq_center_iff.2 rfl
+
+lemma circle_map_ne_center {c : ℂ} {R : ℝ} (hR : R ≠ 0) {θ : ℝ} : circle_map c R θ ≠ c :=
+mt circle_map_eq_center_iff.1 hR
+
+lemma has_deriv_at_circle_map (c : ℂ) (R : ℝ) (θ : ℝ) :
+  has_deriv_at (circle_map c R) (circle_map 0 R θ * I) θ :=
+by simpa only [mul_assoc, one_mul, of_real_clm_apply, circle_map, of_real_one, zero_add]
+ using ((of_real_clm.has_deriv_at.mul_const I).cexp_real.const_mul (R : ℂ)).const_add c
+
+lemma differentiable_circle_map (c : ℂ) (R : ℝ) :
```

Comment:
Couldn't we push to `times_cont_diff ℝ top`?

Reply:
We need to generalize `times_cont_diff.mul` first. Now it doesn't work for multiplication in an algebra over the base field. I'll add a TODO.

Reply:
> We need to generalize `times_cont_diff.mul` first. Now it doesn't work for multiplication in an algebra over the base field.

This has been on my own todo list for a year (it's also what blocks making GL(n) a Lie group).  I can do it at some point, or please ping me if you do it.